### PR TITLE
Support checking remote git tags existence

### PR DESCRIPTION
**Pull request by chuganzy**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

This PR enables to check remote tag existence with `git_tag_exists` action.

### Description

Also switch to use `error_callback` to check existence since `git` commands outputs std logs in some cases like warnings and I thought it's more robust to check using status code.

### CODE REVIEW

File: fastlane/lib/fastlane/actions/git_tag_exists.rb [hunk 1]
```diff
@@ -2,8 +2,19 @@ module Fastlane
   module Actions
     class GitTagExistsAction < Action
       def self.run(params)
-        result = Actions.sh("git rev-parse -q --verify refs/tags/#{params[:tag].shellescape} || true", log: FastlaneCore::Globals.verbose?).chomp
-        !result.empty?
+        tag_ref = "refs/tags/#{params[:tag].shellescape}"
+        if params[:remote]
```
Comment (janpio) [inline fastlane/lib/fastlane/actions/git_tag_exists.rb]: wouldn't it be simpler to just use the existance of `remote_name` to decide if a remote should be checked?

Comment (chuganzy) [reply fastlane/lib/fastlane/actions/git_tag_exists.rb]: Good point!

Comment (chuganzy) [reply fastlane/lib/fastlane/actions/git_tag_exists.rb]: Hmm, but in that case we cannot provide default `remote_name` value..
It's fine to me but most of other `git remote` related actions have default value for `remote`.

Comment (janpio) [reply fastlane/lib/fastlane/actions/git_tag_exists.rb]: Also true, right.
What other actions are there? Quick google only found me https://docs.fastlane.tools/actions/push_git_tags/

Comment (chuganzy) [reply fastlane/lib/fastlane/actions/git_tag_exists.rb]: There is push_to_git_remote I found.
https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/push_to_git_remote.rb

Comment (janpio) [reply fastlane/lib/fastlane/actions/git_tag_exists.rb]: Hmm... problem in comparing those is that both `push_git_tags` and `push_to_git` by default interact with a remote, so having a `remote` param with default value `origin` makes total sense. But this one works locally by default, and only with this change can does remote things, right?

Comment (janpio) [reply fastlane/lib/fastlane/actions/git_tag_exists.rb]: (Which of course doesn't help us solve the problem at all :p)

Comment (chuganzy) [reply fastlane/lib/fastlane/actions/git_tag_exists.rb]: Correct, and it totally makes sense. Had a great discussion here, thanks! 
Will fix it only to use `remote_name`!

Comment (janpio) [reply fastlane/lib/fastlane/actions/git_tag_exists.rb]: That wasn't really my conclusion though. I think you are good with how it is right now if you want to.

**Review by getaaron — APPROVED** [review]

**Comment by nafu** [comment]
@chuganzy Thank you for your contribution ✨

**Pull request merged**